### PR TITLE
[rush] Add rush add support for yarn

### PR DESCRIPTION
**Pull request by ron4ex**
Addresses #1343

### CODE REVIEW

File: apps/rush-lib/src/logic/PackageJsonUpdater.ts [hunk 1]
```diff
@@ -316,14 +312,26 @@ export class PackageJsonUpdater {
...
         Utilities.executeCommandAndCaptureOutput(
           this._rushConfiguration.packageManagerToolFilename,
-          ['view', packageName, 'versions', '--json'],
+          commandArgs,
           this._rushConfiguration.commonTempFolder
         );
 
-      const versionList: Array<string> = JSON.parse(allVersions);
+        let versionList: Array<string>;
+
```
Comment (iclanton) [inline apps/rush-lib/src/logic/PackageJsonUpdater.ts]:
```suggestion
```

File: apps/rush-lib/src/logic/PackageJsonUpdater.ts [hunk 2]
```diff
@@ -316,14 +312,26 @@ export class PackageJsonUpdater {
       console.log();
       console.log(`Querying registry for all versions of "${packageName}"...`);
 
+      let commandArgs: Array<string> = ['view', packageName, 'versions', '--json'];
```
Comment (iclanton) [inline apps/rush-lib/src/logic/PackageJsonUpdater.ts]: Let's put this in an explicit `else` block, similar to the code starting on line 329

File: apps/rush-lib/src/logic/PackageJsonUpdater.ts [hunk 3]
```diff
@@ -346,9 +354,14 @@ export class PackageJsonUpdater {
       }
       console.log(`Querying NPM registry for latest version of "${packageName}"...`);
 
+      let commandArgs: Array<string> = ['view', `${packageName}@latest`, 'version'];
```
Comment (iclanton) [inline apps/rush-lib/src/logic/PackageJsonUpdater.ts]: Same here.

**Comment by iclanton** [comment]
Thanks for putting this together! I just left a few comments regarding style.

File: common/changes/@microsoft/rush/yarn-rush-add_2019-10-13-20-03.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{
+  "changes": [
+    {
+      "packageName": "@microsoft/rush",
+      "comment": "Added rush add support for yarn",
```
Comment (iclanton) [inline common/changes/@microsoft/rush/yarn-rush-add_2019-10-13-20-03.json]:
```suggestion
      "comment": "Add support for \"rush add\" for repos using the Yarn package manager.",
```

**Comment by iclanton** [comment]
@ron4ex - Can you commit the update to the changefile? After that, this should be good to go.

**Comment by octogonz** [comment]
This fix was published with Rush 5.16.0

**Pull request merged**